interfaces/tracing/v2/schema.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2024 Canonical
+# See LICENSE file for licensing details.
+"""This file defines the schemas for the provider and requirer sides of the tracing interface.
+
+It exposes two interfaces.schema_base.DataBagSchema subclasses called:
+- ProviderSchema
+- RequirerSchema
+
+Examples:
+    RequirerSchema:
+        unit_data: <empty>
+        application_data:
+          receivers:
+            - otlp_grpc
+            - otlp_http
+
+    ProviderSchema:
+        # unit_data: <empty>
+        application_data:            
+          receivers:
+            - protocol: 
+                name: otlp_http
+                type: http
+              url: http://traefik_address:2331
+            - protocol: 
+                name: otlp_grpc
+                type: grpc
+              url: traefik_address:2331
+
+"""
+from typing import List
+import enum
+
+from interface_tester.schema_base import DataBagSchema
+from pydantic import BaseModel, Json, Field, ConfigDict
+
+
+class TransportProtocolType(str, enum.Enum):
+    """Receiver Type."""
+    http = "http"
+    grpc = "grpc"
+    
+class ProtocolType(BaseModel):
+    """Protocol Type."""
+    model_config = ConfigDict(
+        # Allow serializing enum values.
+        use_enum_values=True
+    )
+    """Pydantic config."""
+    
+    name: str = Field(
+      ...,
+        description="Receiver protocol name. What protocols are supported (and what they are called) "
+                    "may differ per provider.",
+        examples=["otlp_grpc", "otlp_http", "tempo_http", "jaeger_thrift_compact"])
+    type: TransportProtocolType = Field(
+        ...,
+        description="The transport protocol used by this receiver.",
+        examples=["http", "grpc"])
+    
+class Receiver(BaseModel):
+    """Specification of an active receiver."""
+    
+    protocol: ProtocolType = Field(
+        ...,
+        description="Receiver protocol name and type.")
+    url: str = Field(
+      ...,
+      description="URL at which the receiver is reachable. If there's an ingress, it would be the external URL. Otherwise, it would be the service's fqdn or internal IP",
```

Comment:
let's copy here the comment we have in the charm lib. IIRC it mentioned the absence of the scheme in case it's a grpc endpoint